tsconfig.json

```diff
@@ -0,0 +1,22 @@
+{
+  "compilerOptions": {
+    "target": "ESNext",
+    "lib": ["DOM", "DOM.Iterable", "ESNext"],
+    "allowJs": false,
+    "allowSyntheticDefaultImports": true,
+    "strict": true,
+    "allowUnusedLabels": false,
+    "allowUnreachableCode": false,
+    "noImplicitOverride": true,
+    "noImplicitReturns": true,
+    "noUncheckedIndexedAccess": true,
+    "forceConsistentCasingInFileNames": true,
```

Comment:
Wondering if we need all of these custom config - e.g isn't `forceConsistentCasingInFileNames` already `true` by default? 

Should maybe be configured via `core.ignorecase` in `git` as it also applies to the other files in the repo.

Reply:
I took inspiration from [this](https://github.com/tsconfig/bases/blob/main/bases/strictest.json) `tsconfig` that enables more strict rules to make the code less error-prone, and I removed some unnecessary rules. `forceConsistentCasingInFileNames` is already true by default, so we can remove it as well.